libraries/Microsoft.Bot.Builder/Inspection/InspectionMiddleware.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    public class InspectionMiddleware : InterceptionMiddleware
+    {
+        private InspectionState _inspectionState;
+        private UserState _userState;
+        private ConversationState _conversationState;
+        private MicrosoftAppCredentials _credentials;
+
+        public InspectionMiddleware(InspectionState inspectionState, UserState userState = null, ConversationState conversationState = null, MicrosoftAppCredentials credentials = null, ILogger<InspectionMiddleware> logger = null)
+            : base(logger)
+        {
+            _inspectionState = inspectionState;
+            _userState = userState;
+            _conversationState = conversationState;
+            _credentials = credentials ?? MicrosoftAppCredentials.Empty;
+            HttpClient = new HttpClient();
+        }
+
+        protected virtual HttpClient HttpClient { get; set; }
+
+        public async Task<bool> ProcessCommandAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (turnContext.Activity.Type == ActivityTypes.Message)
+            {
+                var command = turnContext.Activity.Text.Split(' ');
+                if (command.Length > 1 && command[0] == "/DEBUG")
+                {
+                    if (command.Length == 2 && command[1] == "open")
+                    {
+                        await ProcessOpenCommandAsync(turnContext, cancellationToken).ConfigureAwait(false);
+                        return true;
+                    }
+
+                    if (command.Length == 3 && command[1] == "attach")
+                    {
+                        await ProcessAttachCommandAsync(turnContext, command[2], cancellationToken).ConfigureAwait(false);
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        protected override async Task<(bool shouldForwardToApplication, bool shouldIntercept)> InboundAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
```

Comment:
I'm unclear why an "Inspection" middleware class has a "shouldIntercept" return result.

Reply:
because it might be a command